### PR TITLE
Script for granting restricted sql account in RCP

### PR DESCRIPTION
**Pull request by javsanbel2**
When hive metastore is executed in read-only mode, we are creating a restricted SQL account with only SELECT permissions to be used.

When hive metastore is executed in read-write mode, we just use the SQL account provided by our secrets.

### CODE REVIEW

**Review by rpoluri — CHANGES_REQUESTED** [review]
this is already handled in apiary-data-lake terraform here
https://github.com/ExpediaGroup/apiary-data-lake/blob/master/scripts/mysql-user.sh

so we should similary using helm init job in RCP

File: files/startup.sh [hunk 1]
```diff
@@ -170,5 +170,20 @@ if [ ! -z ${ECS_CONTAINER_METADATA_URI} ]; then
...
 
+# Creating user with only SELECT permissions when is read_only hive metastore (credentials from Vault)
+if [ ! -z $MYSQL_USER_CREDS ] && [ "$HIVE_METASTORE_ACCESS_MODE" = "readonly" ]; then
+
+    # Get credentials from vault (HM_MYSQL_USER_READONLY_USERNAME & HM_MYSQL_USER_READONLY_PASSWORD)
+    if [[ -f /vault/secrets/secrets.txt ]]; then
+        source /vault/secrets/secrets.txt
+    fi
+
+    /allow-grant-rcp.sh
```
Comment (barnharts4) [inline files/startup.sh:181]: Additionally, wherever the right change ends up, since this is open-source, we need to name and implement things in a way that is not focused on technologies or services that are internal to EG.

RCP is completely an EG concept, and Vault is not/should not be necessary to run Apiary.

We should do something more like:
```
if [[ -f ${EXTERNAL_CREDENTIALS_FILE} ]]; then
  source ${EXTERNAL_CREDENTIALS_FILE}
fi

...
```

Comment (javsanbel2) [reply files/startup.sh]: You are totally right, Scott, would you take another look at the feature pls?

**Comment by javsanbel2** [comment]
> this is already handled in apiary-data-lake terraform here @rpoluri 

Okay, so I will remove the script from here, and I will try to add it as a helm pre-hook in our deployment place, is that okay? thank you

File: files/startup.sh [hunk 2]
```diff
@@ -7,6 +7,15 @@ set -x
...
 
+# Setting custom credentials from a external file, useful for accounts with limited access.
+if [ -z $EXTERNAL_CREDENTIALS_FILE ]; then
+
+    source $EXTERNAL_CREDENTIALS_FILE
+    
+    export MYSQL_DB_USERNAME=$(echo ${!HM_CUSTOM_USERNAME_VAR_NAME})
+    export MYSQL_DB_PASSWORD=$(echo ${!HM_CUSTOM_PASSWORD_VAR_NAME})
+    echo "Read-only Hive metastore using custom credentials from external file under $MYSQL_DB_USERNAME"
+fi
```
Comment (barnharts4) [inline files/startup.sh:18]: 1. If you are wanting to use external credentials instead of reading from AWS secrets manager, then the check to use them must come *before* lines 6-7 where we try to read from AWS secrets manager.  Otherwise we will always be issuing calls to secrets manager which will fail in RCP.
2. Using Secrets manager or external credentials should be an either/or situation, so you want this in an "if-else".  Currently it runs one path, then the other.
3. `-z` tests if a shell variable is *UNDEFINED*.  So your current code is trying to source the `EXTERNAL_CREDENTIALS_FILE` if the variable is *undefined*, which is the opposite of what you want.
4. Use `[[ ]]` instead of `[ ]` to maintain consistency with our script code.
5. Please put `$EXTERNAL_CREDENTIALS_FILE` in double quotes in the `test` statement.  Not having it in quotes can cause errors in certain edge cases when the variable is undefined or zero-length.
6. Since you are using bash indirection syntax here (`${!HM_CUSTOM_USERNAME_VAR_NAME}`), it is unclear what the exact format/syntax of a valid `EXTERNAL_CREDENTIALS_FILE` should be.  This needs to be added to the documentation in `README.md` in the PR.

**Review by barnharts4 — CHANGES_REQUESTED** [review]

**Comment by javsanbel2** [comment]
Closing this since it won't be needed for now, thanks for the comments
